pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -1711,4 +1725,14 @@ String moveSegmentFile(String rawTableName, String segmentName, String segmentLo
   URI createSegmentPath(String rawTableName, String segmentName) {
     return URIUtils.getUri(_controllerConf.getDataDir(), rawTableName, URIUtils.encode(segmentName));
   }
+
+  public Map<String, SegmentErrorInfo> getSegmentErrors(String tableNameWithType) {
```

Comment:
Why not storing a 2 level map: `Map<String, Map<String, SegmentErrorInfo>>`

Reply:
Hmm - I was following the example of `protected LoadingCache<Pair<String, String>, SegmentErrorInfo> _errorCache;` but the same constraint is not here. I'll change it.

Reply:
One issue with a 2-level map is that it is harder to control the size. Getting the total size of the errors will require iteration.